packages/ag-charts-enterprise/src/series/treemap/treemapSeries.ts

```diff
@@ -406,55 +463,109 @@ export class TreemapSeries extends _ModuleSupport.HierarchySeries<TreemapNodeDat
-        colorScale.domain = colorDomain;
-        colorScale.range = colorRange;
-        colorScale.update();
+        type CreateTreeNodeDatumParams = {
+            datum?: TreeDatum;
+            depth?: number;
+            parent?: TreemapNodeDatum;
+            children?: TreeDatum[];
+        };
+        const createTreeNodeDatum = ({
+            datum,
+            depth,
+            parent,
+            children = datum != null ? datum[childrenKey] : undefined,
+        }: CreateTreeNodeDatumParams) => {
+            const isLeaf = children == null || children.length === 0;
+
+            const labelStyle = isLeaf ? tile.label : group.label;
+            let label: string | undefined;
+            if (datum != null && depth != null && labelKey != null && labelStyle.enabled) {
+                label = this.getLabelText(labelStyle, {
+                    datum,
+                    depth,
+                    parent: parent?.datum,
+                    childrenKey,
+                    colorKey,
+                    labelKey,
+                    secondaryLabelKey,
+                    sizeKey,
+                    value: datum[labelKey] ?? '',
+                });
+            }
 
-        const createTreeNodeDatum = (datum: TreeDatum, depth = 0, parent?: TreemapNodeDatum) => {
-            let label;
-            if (labelFormatter) {
-                label = this.ctx.callbackCache.call(labelFormatter, {
+            let secondaryLabel: string | undefined;
+            if (isLeaf && datum != null && depth != null && secondaryLabelKey != null && tile.secondaryLabel.enabled) {
+                secondaryLabel = this.getLabelText(tile.secondaryLabel, {
                     datum,
-                    seriesId: this.id,
-                    value: null,
+                    depth,
+                    parent: parent?.datum,
+                    childrenKey,
+                    colorKey,
+                    labelKey,
+                    secondaryLabelKey,
+                    sizeKey,
+                    value: datum[secondaryLabelKey] ?? '',
                 });
             }
-            label ??= labelKey ? datum[labelKey] ?? '' : '';
-            let colorScaleValue = colorKey ? datum[colorKey] ?? depth : depth;
+
+            let colorScaleValue = (datum != null && colorKey != null ? datum[colorKey] : undefined) ?? depth;
             colorScaleValue = validateColor(colorScaleValue);
-            const isLeaf = !datum.children;
-            let fill = groupFill;
+            let fill = isLeaf ? tile.fill : group.fill;
             if (typeof colorScaleValue === 'string') {
                 fill = colorScaleValue;
-            } else if (isLeaf || !groupFill) {
+            } else if (colorScale != null) {
                 fill = colorScale.convert(colorScaleValue);
             }
+
             const nodeDatum: TreemapNodeDatum = {
-                datum,
+                datum: datum!,
```

Comment:
Nit: Can we avoid the use of `!`? Is there really a case for `createTreeNodeDatum()` without a datum?